Framework/DataHandling/src/LoadDNSEvent.cpp

```diff
@@ -0,0 +1,479 @@
+  static const auto skipTable = buildSkipTable2(block_sep);
+
+  const uint64_t minChunckSize = MAX_BUFFER_BYTES_SIZE;
+  const uint64_t chunckSize = std::max(minChunckSize, file.fileSize() / maxChunckCount);
+
+  std::vector<std::vector<uint8_t>> result;
+
+  while (!file.eof()) {
+    result.push_back(std::vector<uint8_t>());
+    // read a big chunck of file:
+    auto &data = result.back();
+    data.resize(chunckSize);
+    try {
+      file.readRaw(*data.begin(), chunckSize);
+    } catch (std::ifstream::failure &) {
+      data.resize(file.gcount());
+      return result;
+    }
+
+    // search for a block_separator, and append everything up to it :
+    static const auto windowSize = block_sep.size();
+    uint8_t *current_window = nullptr;
+    std::array<uint8_t, windowSize> *windowAsArray =
+        reinterpret_cast<std::array<uint8_t, windowSize> *>(current_window);
+
+    try {
+      data.resize(data.size() + windowSize);
+      // accomodate for possible relocation of vector...:
+      current_window = &*(data.end() - windowSize);
+      windowAsArray = reinterpret_cast<std::array<uint8_t, windowSize> *>(current_window);
+      file.readRaw(current_window[0], windowSize);
+
+      while ((*windowAsArray != block_sep) & (!file.eof())) {
+        size_t skip_length = skipTable[current_window[windowSize - 1]];
+
+        const auto orig_data_size = data.size();
+        data.resize(orig_data_size + skip_length);
+
+        // accomodate for possible relocation of vector...:
+        current_window = (&data.back() - windowSize + 1);
+
+        windowAsArray = reinterpret_cast<std::array<uint8_t, windowSize> *>(current_window);
+        file.readRaw(current_window[windowSize - skip_length], skip_length);
+      }
+    } catch (std::ifstream::failure &) {
+      return result;
+    }
+  } // while
+  return result;
+}
+
+LoadDNSEvent::EventAccumulator LoadDNSEvent::parse_File(FileByteStream &file, const std::string &fileName) {
+  // File := Header Body
+  std::vector<uint8_t> header = parse_Header(file);
+
+  // check it is actually a mesytec psd listmode file:
+  if (!startsWith(header, std::string("mesytec psd listmode data"))) {
+    g_log.error() << "This seems not to be a mesytec psd listmode file: " << fileName;
+    throw Exception::FileError("This seems not to be a mesytec psd listmode file: ", fileName);
+  }
+
+  // Parse actual data:
+  const int threadCount = PARALLEL_GET_MAX_THREADS;
+  // Split File:
+  std::vector<std::vector<uint8_t>> filechuncks = split_File(file, threadCount);
+  g_log.debug() << "filechuncks count = " << filechuncks.size() << std::endl;
+
+  std::vector<EventAccumulator> eventAccumulators(filechuncks.size());
+  for (auto &evtAcc : eventAccumulators) {
+    evtAcc.neutronEvents.resize(m_detectorPixelCount);
+  }
+
+  // parse individual file chuncks:
+  PARALLEL_FOR_IF(true)
+  for (int i = 0; i < static_cast<int>(filechuncks.size()); ++i) {
```

Comment:
Can we make `int i` into `size_t i` then all the static casts go away

Reply:
oh jeah, that's nice

Reply:
or, that is not working, it says the index variable for OpenMP must be a signed integer

Reply:
Ah yes I forgot MSVC has an old OpenMP standard :face_exhaling: